### DISCUSSION TITLE
containerMode:Kubernetes. Use Empty dir

### DISCUSSION THREAD

**Discussion by DenisPalnitsky** [actions/actions-runner-controller#3318] [Questions]
I wonder if there is a way to use `emptyDir` instead of provisioning volume for every workflow.
It seems that currently, the workflow runner relies on PVC however I cannot see the reason why it could not use `emptyDir`.

Provisioning storage is problematic in our case as we have spikes in workflow jobs that could  spin up over a 1000 extra pods and provisioning storage for each of them becomes a performance bottleneck.

**Comment by DenisPalnitsky** [comment]
I think I find an answer after some digging. The `runner` and `runner-workflow` should use same volume because `runner` copies the scripts that will be executed by `runner-workflow`

**Reply by rteeling-evernorth** [reply]
This isn't possible. Emptydir volumes cannot be shared between separate pods, they are explicitly scoped to the lifecycle of a pod.

https://kubernetes.io/docs/concepts/storage/volumes/#emptydir
